test/HypLSP7.t.sol

```diff
@@ -0,0 +1,299 @@
+
+        remoteToken = new HypLSP7(DECIMALS, address(remoteMailbox));
+
+        remoteToken.initialize(TOTAL_SUPPLY, NAME, SYMBOL, address(noopHook), address(0), OWNER);
+
+        vm.prank(OWNER);
+        remoteToken.enrollRemoteRouter(ORIGIN, address(localToken).addressToBytes32());
+
+        igp = new TestInterchainGasPaymaster();
+
+        vm.deal(ALICE, 125_000);
+    }
+
+    function _enrollRemoteTokenRouter() internal {
+        vm.prank(OWNER);
+        remoteToken.enrollRemoteRouter(ORIGIN, address(localToken).addressToBytes32());
+    }
+
+    function _expectRemoteBalance(address _user, uint256 _balance) internal {
+        assertEq(remoteToken.balanceOf(_user), _balance);
+    }
+
+    function _processTransfers(address _recipient, uint256 _amount) internal {
+        vm.prank(address(remoteMailbox));
+        remoteToken.handle(
+            ORIGIN, address(localToken).addressToBytes32(), abi.encodePacked(_recipient.addressToBytes32(), _amount)
+        );
+    }
+
+    function _setCustomGasConfig() internal {
+        vm.prank(OWNER);
+        localToken.setHook(address(igp));
+
+        TokenRouter.GasRouterConfig[] memory config = new TokenRouter.GasRouterConfig[](1);
+        config[0] = GasRouter.GasRouterConfig({ domain: DESTINATION, gas: GAS_LIMIT });
+
+        vm.prank(OWNER);
+        localToken.setDestinationGas(config);
+    }
+
+    function _performRemoteTransfer(uint256 _msgValue, uint256 _amount) internal {
+        vm.prank(ALICE);
+
+        localToken.transferRemote{ value: _msgValue }(DESTINATION, BOB.addressToBytes32(), _amount);
+
+        vm.expectEmit(true, true, false, true);
+
+        emit ReceivedTransferRemote(ORIGIN, BOB.addressToBytes32(), _amount);
+        _processTransfers(BOB, _amount);
+
+        assertEq(remoteToken.balanceOf(BOB), _amount);
+    }
+
+    function _performRemoteTransferAndGas(uint256 _msgValue, uint256 _amount, uint256 _gasOverhead) internal {
+        uint256 ethBalance = ALICE.balance;
+
+        _performRemoteTransfer(_msgValue + _gasOverhead, _amount);
+
+        assertEq(ALICE.balance, ethBalance - REQUIRED_VALUE - _gasOverhead);
+    }
+
+    function _performRemoteTransferWithEmit(uint256 _msgValue, uint256 _amount, uint256 _gasOverhead) internal {
+        vm.expectEmit(true, true, false, true);
+        emit SentTransferRemote(DESTINATION, BOB.addressToBytes32(), _amount);
+        _performRemoteTransferAndGas(_msgValue, _amount, _gasOverhead);
+    }
+
+    function testBenchmark_overheadGasUsage() public {
+        vm.prank(address(localMailbox));
+
+        uint256 gasBefore = gasleft();
+        localToken.handle(
+            DESTINATION, address(remoteToken).addressToBytes32(), abi.encodePacked(BOB.addressToBytes32(), TRANSFER_AMT)
+        );
+        uint256 gasAfter = gasleft();
+    }
+}
+
+contract HypLSP7Test is HypTokenTest {
+    using TypeCasts for address;
+
+    HypLSP7 internal lsp7Token;
+
+    address internal owner = makeAddr("owner");
```

Comment:
This address is defined twice. Suggest removing this state variable and using the one above

<img width="478" alt="image" src="https://github.com/user-attachments/assets/5067bf9f-69c4-4839-9e1f-67a3958ff2e5">